src/Controller/Component/PrgComponent.php

```diff
@@ -53,6 +53,35 @@ public function startup()
         return $this->_registry->getController()->redirect($url);
     }
 
+    /**
+     * Populates the $_isSearch view variable based on the current request.
+     *
+     * You need to configure the modelClass config if you are not using the controller's
+     * default modelClass property.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function beforeRender()
+    {
+        $controller = $this->_registry->getController();
+        $modelClass = $this->getConfig('modelClass', $controller->modelClass);
+        if (!$modelClass) {
+            return null;
+        }
+
+        if (!isset($controller->{$modelClass})) {
```

Comment:
For plugins value of `$controller->modelClass` will be of form `Foo.Bar` so this check will fail. You'll need to use `pluginSplit()` first.

Reply:
Ah right, thanks.

Reply:
Actually an even better way to get model instance would be using:
```php
$modelClass = $this->getConfig('model');
$model = $controller->loadModel($modelClass);
```
If `$modelClass` is `null` the controller's default model will be automatically returned.

Reply:
If the model at this point is not loaded yet, I assume it should not be loaded either (as the value returned must be logically false anyway - as no search query has been done at all).

Reply:
If `$modelClass` is same as `Controller::$modelClass`, which is going to be the case most of the times, `$controller->{$modelClass}` is always gonna trigger model loading. But okay we can keep current code.